### PR TITLE
Add some system level tests based on examples for WLAN.

### PR DESCRIPTION
**Pull request by reckenro**
### What does this Pull Request accomplish?

Adds four system level tests based on some of the WLAN C Examples.

Related work item: AB#1794514.

### Why should this Pull Request be merged?

We want to spot check the WLAN service and doing so with examples seems like a good way.

### What testing has been done?

WLAN Driver Api system tests pass:

image

### Up next for WLAN

I think we need to go through the AC for the story to determine if we want any more example based tests, general tests, or if the remaining should be manual spot check on concerning methods / params + reliance on SpecAn/Instr tests covering similar patterns.

### CODE REVIEW

File: source/tests/system/nirfmxwlan_driver_api_tests.cpp
Comment (reckenro) [inline source/tests/system/nirfmxwlan_driver_api_tests.cpp:19]: I"m still trying to figure this out, but the TXP related tests gave this warning but still got results. The full message was "379206: Rising edge detection failed for the acquired waveform.". I'm trying to figure out if this is problematic or not.

Comment (astarche) [reply source/tests/system/nirfmxwlan_driver_api_tests.cpp:19]: I think it's OK. I've hit the same type of warning running on hardware that's not set up with the right cables/external-config etc.

Also can you use `UPPER_SNAKE` for constants, at least in files that don't already use `kWhatever`. I know I've been ignoring the session tests ...

File: source/tests/system/nirfmxwlan_driver_api_tests.cpp
Comment (astarche) [inline source/tests/system/nirfmxwlan_driver_api_tests.cpp:156]: Some of these field names will be broken by David's change right? Or is wlan not included in that yet?

File: source/tests/system/nirfmxwlan_driver_api_tests.cpp
Comment (astarche) [inline source/tests/system/nirfmxwlan_driver_api_tests.cpp:149]: These are declared separately because their codegenerated? I'll allow it. That's fine.

File: source/tests/system/nirfmxwlan_driver_api_tests.cpp
Comment (astarche) [inline source/tests/system/nirfmxwlan_driver_api_tests.cpp:355]: Nice chunk of tests!

**Review by astarche — APPROVED** [review]
Nice chunk of tests!

File: source/tests/system/nirfmxwlan_driver_api_tests.cpp
Comment (reckenro) [reply source/tests/system/nirfmxwlan_driver_api_tests.cpp:156]: Yeah, you're right. I think his is getting close so I'll wait for #499 to go in and then merge main and fix up the names.

Comment (DavidCurtiss) [reply source/tests/system/nirfmxwlan_driver_api_tests.cpp:156]: I can also regen after the update, so Ryan can merge that in rather than doing it manually.

File: source/tests/system/nirfmxwlan_driver_api_tests.cpp
Comment (DavidCurtiss) [reply source/tests/system/nirfmxwlan_driver_api_tests.cpp:149]: I intentionally generated variable declarations at the top of the block in which they're first used. If you'd rather they all be at the top of the function, I can do that.
(The examples define all the variables at the top of the function because they're in C [vs. C++], which requires that.)

Comment (astarche) [reply source/tests/system/nirfmxwlan_driver_api_tests.cpp:149]: Oh no, this is way better than the C examples! That style is a nightmare for transcription for one thing.

My preference is:

```c++
const auto ofdm_mod_acc_fetch_composite_rmsevm_response = client::ofdm_mod_acc_fetch_composite_rmsevm(stub(), session, "", 10.0);
```

That allows it to be const and all-on-one-line. I think of it as avoiding the extra default initialization, but that might be optimizer-dependent anyway (though some of that is more official in c++17).

Comment (DavidCurtiss) [reply source/tests/system/nirfmxwlan_driver_api_tests.cpp:149]: Ah, right. I agree. I had another look at the generator, and it was easy enough to change. @reckenro I pushed another change that can be merged in, to inline the declarations when possible. You (or I) can probably just include it in a future update.

File: source/tests/system/nirfmxwlan_driver_api_tests.cpp
Comment (DavidCurtiss) [reply source/tests/system/nirfmxwlan_driver_api_tests.cpp:156]: Done. You can find the updates at the usual place, @reckenro -- also rebased on top of main.

Comment (reckenro) [reply source/tests/system/nirfmxwlan_driver_api_tests.cpp]: Updated from your branch. Got the updates for the acronym names and the field names on one line.

**Pull request merged**
